src/components/ui/task.tsx

```diff
@@ -77,196 +104,207 @@ const TaskItem: React.FC<TaskItemProps> = ({ item, isActive, isVisible }) => {
+                        size={18}
+                        color="#888"
+                      />
+                      <Text className="ml-1 text-sm text-gray-600">
+                        Notes: {item.notes}
+                      </Text>
+                    </View>
+                  )}
+                  {item.group && (
+                    <View className="flex-row items-center">
+                      <Ionicons name="people-outline" size={18} color="#888" />
+                      <Text className="ml-1 text-sm text-gray-600">
+                        Group: {item.group}
+                      </Text>
+                    </View>
+                  )}
+                </View>
+              )}
+            </TouchableOpacity>
+
+            <TouchableOpacity
+              className="px-2"
+              onPress={() => onThreeDotsPress?.(item)}
+            >
+              <Ionicons name="ellipsis-horizontal" size={24} color="black" />
+            </TouchableOpacity>
 
-          <TouchableOpacity className={'px-2'}>
-            <Ionicons name="ellipsis-horizontal" size={24} color="black" />
+            <Animated.View
+              style={[
+                cn('absolute -right-4 w-3'),
+                animatedHeightStyle,
+                { backgroundColor: item.groupColor },
+              ]}
+            />
+          </Animated.View>
+        </PanGestureHandler>
+      ) : (
+        <View className="flex-row items-center justify-between h-full rounded-lg bg-white px-2 py-1">
+          <View className="relative w-6 h-6 justify-center items-center mr-2">
+            {item.icon}
+          </View>
+
+          <View className="flex-1 pr-2">
+            <Text
+              className="text-sm font-bold text-black"
+              numberOfLines={1}
+              ellipsizeMode="tail"
+            >
+              {item.title}
+            </Text>
+            <Text
+              className="text-gray-600 text-xs"
+              numberOfLines={1}
+              ellipsizeMode="tail"
+            >
+              {calculateTimeLeft(item.dueDate)}
+            </Text>
+          </View>
+
+          <TouchableOpacity
+            className="px-1"
+            onPress={() => onThreeDotsPress?.(item)}
+          >
+            <Ionicons name="ellipsis-horizontal" size={18} color="black" />
           </TouchableOpacity>
 
           <Animated.View
             style={[
-              cn('absolute right--4 w-[12px]'),
+              cn('absolute -right-4 w-3'),
               animatedHeightStyle,
-              { backgroundColor: item.groupColor },
+              {
+                backgroundColor: item.groupColor,
+              },
             ]}
           />
-        </Animated.View>
-      </PanGestureHandler>
+        </View>
+      )}
 
-      <Animated.View style={DoneBoxStyle}>
-        <Ionicons name="checkmark" size={24} color="white" />
-      </Animated.View>
+      {!isCompact && (
+        <>
```

Comment:
wtf is this ?

Reply:
Used it because in other ways delete and done boxes will be under task, didn't find other way to replace it